src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -203,14 +203,143 @@ pub struct PCIeRootComplex {
     pub root_bus_devices: HashMap<String, PCIeEndpoint>,
 }
 
-#[derive(Debug, Default)]
+/// PCIePortBusPrefix defines the naming scheme for PCIe ports.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum PCIePortBusPrefix {
+    RootPort,
+    SwitchPort,
+    SwitchUpstreamPort,
+    SwitchDownstreamPort,
+}
+
+impl std::fmt::Display for PCIePortBusPrefix {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let prefix = match self {
+            PCIePortBusPrefix::RootPort => "rp",
+            PCIePortBusPrefix::SwitchPort => "sw",
+            PCIePortBusPrefix::SwitchUpstreamPort => "swup",
+            PCIePortBusPrefix::SwitchDownstreamPort => "swdp",
+        };
+        write!(f, "{}", prefix)
+    }
+}
+
+/// PCIePort distinguishes between different types of PCIe ports.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash)]
+pub enum PCIePort {
+    // NoPort is for disabling VFIO hotplug/coldplug
+    #[default]
+    NoPort,
+
+    /// RootPort attach VFIO devices to a root-port
+    RootPort,
+
+    // SwitchPort attach VFIO devices to a switch-port
+    SwitchPort,
+
+    // InvalidPort is for invalid port
+    InvalidPort,
```

Comment:
I don't see any usage of `InvalidPort`, perhaps it could be removed?

Reply:
fixed

Reply:
Thanks!